### PR TITLE
antlr4 cpp runtime library

### PR DESCRIPTION
**Pull request by NineKa**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----
antlr4 cpp runtime library

### CODE REVIEW

**Comment by fxcoudert** [comment]
I'm not sure about the name for this formula (`antlr4-cpp-runtime`). Why include version number in the name? How does upstream call it? (the project / source base, not the tarball itself)

**Comment by NineKa** [comment]
the main header file provided is called `antlr4-runtime.h`. I guess they want to distinguish from the cpp runtime library built for antlr3.

The source code is maintained at https://github.com/antlr/antlr4/tree/master/runtime/Cpp

File: Formula/antlr4-cpp-runtime.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
+class Antlr4CppRuntime < Formula
+  desc "ANother Tool for Language Recognition C++ Runtime Library"
+  homepage "http://www.antlr.org"
+  url "http://www.antlr.org/download/antlr4-cpp-runtime-4.7-source.zip"
+  sha256 "3aa4bac23c60df14a687839b5e6aa7e94054112d3d3c5c8b1cffe270a4aeeaf7"
+
+  depends_on "cmake" => :build
+  depends_on "antlr" => :recommended
```
Comment (fxcoudert) [inline Formula/antlr4-cpp-runtime.rb]: Is this actually usable without `antlr`?

**Comment by NineKa** [comment]
moved to https://github.com/Homebrew/homebrew-core/pull/13738

**Pull request closed without merging**